src/screens/UserManagement/UserRevamp/DropdownWithLoading.res

```diff
@@ -0,0 +1,160 @@
+open HeadlessUI
+
+type dropDownState = Loading | Success | NoData
+
+let commonDropdownCss = "absolute md:max-h-36 md:min-h-36 overflow-scroll z-30 w-full bg-white rounded-sm shadow-lg  focus:outline-none my-1 border border-jp-gray-lightmode_steelgray border-opacity-75  ring-1 ring-black ring-opacity-5"
+
+module DropDownItems = {
+  @react.component
+  let make = (
+    ~options: array<SelectBox.dropdownOption>,
+    ~formKey,
+    ~keyValueFromForm,
+    ~setArrow,
+  ) => {
+    let form = ReactFinalForm.useForm()
+    let onItemSelect = value => {
+      form.change(formKey, value->Identity.genericTypeToJson)
+    }
+    <Menu.Items className={`divide-y divide-gray-100 ${commonDropdownCss}`}>
+      {props => {
+        if props["open"] {
+          setArrow(_ => true)
```

Comment:
adding setState in the render will cause a issue

Reply:
Removed it